lib/bson/ext_json.rb

```diff
@@ -139,7 +139,7 @@ module ExtJSON
         return {}
       end
 
-      if hash.key?('$ref')
+      if hash['$ref']&.is_a?(String) && hash.key?('$id')
```

Comment:
For context, from the dbref spec: 
> Assert that each document is not decoded to a DBRef model. In the context of implicit decoding, the document MUST be decoded like any other embedded document. In the context of explicit decoding, the DBRef decoding method MUST raise an error.

Reply:
I understand implicit decoding to be this block of code, and explicit decoding to be the changes made in the constructor for dbref.

Reply:
After speaking with Oleg, this is the correct understanding... Resolving.